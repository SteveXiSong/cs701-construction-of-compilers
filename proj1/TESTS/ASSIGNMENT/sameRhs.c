#include <stdio.h>

int main() {
  int g1, g2;
  int a, b, c, w, x, y, z;
  scanf("%d", &g1);
  g2 = g1;
  a = g2 + 33;
  scanf("%d", &x);  /* other instruction (in same basic block) */
  b = g2 + 33;      /* redundant */
  scanf("%d", &y);  /* other instruction (in same basic block) */
  c = g2 + 33;      /* redundant */
  scanf("%d", &z);  /* other instruction (in same basic block) */
  g1 = a+b+c;
  printf("g1: %d\ng2: %d\na: %d\nb: %d\nc: %d\n", g1, g2, a, b, c);
}
