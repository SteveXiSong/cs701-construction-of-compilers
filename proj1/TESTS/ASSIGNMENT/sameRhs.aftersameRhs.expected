
FUNCTION main

BASIC BLOCK entry
%1:	alloca	 XXX
%2:	alloca	 XXX
%3:	alloca	 XXX
%4:	alloca	 XXX
%5:	call	 XXX %1 __isoc99_scanf
%6:	load	 %1
%7:	add	 %6 XXX
%8:	call	 XXX %2 __isoc99_scanf
%9:	call	 XXX %3 __isoc99_scanf
%10:	call	 XXX %4 __isoc99_scanf
%11:	add	 %7 %7
%12:	add	 %11 %7
%13:	store	 %12 %1
%14:	load	 %1
%15:	call	 XXX %14 %6 %7 %7 %7 printf
%16:	ret	 XXX
